1_fetch/src/get_nwis_data.R

```diff
@@ -1,48 +1,42 @@
+download_nwis_data <- function(site_num, parameterCd, startDate, endDate){
 
-download_nwis_data <- function(site_nums = c("01427207", "01432160", "01436690", "01466500")){
-  
-  # create the file names that are needed for download_nwis_site_data
-  # tempdir() creates a temporary directory that is wiped out when you start a new R session; 
-  # replace tempdir() with "1_fetch/out" or another desired folder if you want to retain the download
-  download_files <- file.path(tempdir(), paste0('nwis_', site_nums, '_data.csv'))
-  data_out <- data.frame()
-  # loop through files to download 
-  for (download_file in download_files){
-    download_nwis_site_data(download_file, parameterCd = '00010')
-    # read the downloaded data and append it to the existing data.frame
-    these_data <- read_csv(download_file, col_types = 'ccTdcc')
-    data_out <- bind_rows(data_out, these_data)
+  data_out <- readNWISdata(sites = site_num, parameterCd = parameterCd, startDate = startDate, endDate = endDate, service="iv")
+
+  # -- simulating a failure-prone web-service here, do not edit --
+  set.seed(Sys.time())
+  if (sample(c(T,F,F,F), 1)){
+    stop(site_num, ' has failed due to connection timeout. Try tar_make() again')
   }
+    
   return(data_out)
+  
 }
 
-nwis_site_info <- function(fileout, site_data){
-  site_no <- unique(site_data$site_no)
-  site_info <- dataRetrieval::readNWISsite(site_no)
-  write_csv(site_info, fileout)
-  return(fileout)
-}
-
-
-download_nwis_site_data <- function(filepath, parameterCd = '00010', startDate="2014-05-01", endDate="2015-05-01"){
+## Combine nwis datasets 
+combine_nwis_data <- function(...){
+  
+  site_data <- bind_rows(...)
   
-  # filepaths look something like directory/nwis_01432160_data.csv,
-  # remove the directory with basename() and extract the 01432160 with the regular expression match
-  site_num <- basename(filepath) %>% 
-    stringr::str_extract(pattern = "(?:[0-9]+)")
+  return(site_data)
   
-  # readNWISdata is from the dataRetrieval package
-  data_out <- readNWISdata(sites=site_num, service="iv", 
-                           parameterCd = parameterCd, startDate = startDate, endDate = endDate)
+}
 
-  # -- simulating a failure-prone web-sevice here, do not edit --
-  set.seed(Sys.time())
-  if (sample(c(T,F,F,F), 1)){
-    stop(site_num, ' has failed due to connection timeout. Try tar_make() again')
+## Save the full downloaded dataset if desired  
+save_nwis_data_csv <- function(Save_as_csv = F, data, fileout = "1_fetch/out/nwis_data.csv"){
```

Comment:
wondering if there is a way to not require a input for the `data` and the `fileout` args here if save_as_csv is F.

Reply:
not default no longer set. rm `= F`

Reply:
Hi Margaux - I think maybe this change didn't get committed? I checked it out and the default value of `F` is still specified.

Looking at this makes me realize that I made a mistake! The way that I suggested you set this up, the pipeline will fail if `save_as_csv` is `F` and the pipeline has not run before. This is because it's a target with `format = "file"` and so if the file listed in its return value doesn't exist it is considered an error. I apologize for misleading you!

I would suggest that for now you just remove the `Save_as_csv` option altogether, and have your `site_data_csv` target run no matter what. Getting into conditional flow right now might be too much of a sidetrack.

Reply:
alrighty - updated but just remove the conditional binary in `save_data_csv()`. If collaborator does not want to save locally, can disable the target (comment out/remove).